src/function/scalar/list/contains_or_position.cpp

```diff
@@ -14,77 +14,30 @@ inline bool ValueCompare(const string_t &left, const string_t &right) {
 	return StringComparisonOperators::EqualsOrNot<false>(left, right);
 }
 
-template <>
-inline bool ValueCompare(const Value &left, const Value &right) {
-	return left == right;
-}
-
-template <class T>
-static void TemplatedListContainsFunction(DataChunk &args, ExpressionState &state, Vector &result) {
-	D_ASSERT(args.ColumnCount() == 2);
-	auto count = args.size();
-	Vector &list = args.data[0];
-	Vector &value_vector = args.data[1];
-
-	result.SetVectorType(VectorType::FLAT_VECTOR);
-	auto result_entries = FlatVector::GetData<bool>(result); // Create a vector of bool
-	auto &result_validity = FlatVector::Validity(result);
-
-	if (list.GetType().id() == LogicalTypeId::SQLNULL) {
-		result_validity.SetInvalid(0);
-		return;
+struct ContainsFunctor {
+	template <bool IS_INIT = false>
+	static inline bool UpdateResultEntries(idx_t child_idx) {
+		return (IS_INIT) ? false : true;
 	}
+};
 
-	auto list_size = ListVector::GetListSize(list);
-	auto &child_vector = ListVector::GetEntry(list);
-
-	VectorData child_data;
-	child_vector.Orrify(list_size, child_data);
-
-	VectorData list_data;
-	list.Orrify(count, list_data);
-	auto list_entries = (list_entry_t *)list_data.data;
-
-	VectorData value_data;
-	value_vector.Orrify(count, value_data);
-
-	for (idx_t i = 0; i < count; i++) {
-		auto list_index = list_data.sel->get_index(i);
-		auto value_index = value_data.sel->get_index(i);
-
-		if (!list_data.validity.RowIsValid(list_index) || !value_data.validity.RowIsValid(value_index)) {
-			result_validity.SetInvalid(i);
-			continue;
-		}
-
-		const auto &list_entry = list_entries[list_index];
-		auto source_idx = child_data.sel->get_index(list_entry.offset);
-		auto child_value = FlatVector::GetData<T>(child_vector);
-
-		auto values = FlatVector::GetData<T>(value_vector);
-		result_entries[list_index] = false;
-		for (idx_t child_idx = 0; child_idx < list_entry.length; child_idx++) {
-			auto child_value_idx = source_idx + child_idx;
-
-			if (!child_data.validity.RowIsValid(child_value_idx)) {
-				continue;
-			}
-			if (ValueCompare<T>(child_value[child_value_idx], values[value_index])) {
-				result_entries[list_index] = true;
-				break; // Found value in list, no need to look further
-			}
-		}
+struct PositionFunctor {
+	template <bool IS_INIT = false>
+	static inline int32_t UpdateResultEntries(idx_t child_idx) {
+		return (IS_INIT) ? 0 : child_idx + 1;
```

Comment:
Instead of having to check `IS_INIT` on every call to `UpdateResultEntries`, maybe you could add an `Initialize()` to your functors, which you call once.